pkg/controller/infrastructure/infraflow/reconcile.go

```diff
@@ -1382,3 +1452,30 @@ func (c *FlowContext) getZone(item *awsclient.Subnet) *aws.Zone {
 func getZoneName(item *awsclient.Subnet) string {
 	return item.AvailabilityZone
 }
+
+func cidrSubnet(baseCIDR string, newPrefixLength int, index int) (string, error) {
+	_, ipNet, err := net.ParseCIDR(baseCIDR)
+	if err != nil {
+		return "", err
+	}
+
+	baseIP := ipNet.IP
+	maskSize, addrSize := ipNet.Mask.Size()
+
+	if newPrefixLength <= maskSize || newPrefixLength > addrSize {
+		return "", fmt.Errorf("invalid new prefix length")
+	}
+
+	subnetSize := big.NewInt(1)
+	subnetSize = subnetSize.Lsh(subnetSize, uint(addrSize-newPrefixLength))
+	offset := big.NewInt(int64(index))
+	offset = offset.Mul(offset, subnetSize)
```

Comment:
Let's make it concise.

```suggestion
	offset := offset.Mul(big.NewInt(int64(index)), subnetSize)
```